crates/sqlexec/src/remote/planner.rs

```diff
@@ -86,3 +92,89 @@ impl PhysicalPlanner for RemotePhysicalPlanner {
         )
     }
 }
+
+#[derive(Clone)]
+pub struct CustomPhysicalPlanner<'a> {
+    ctx: &'a SessionContext,
+}
+
+impl<'a> CustomPhysicalPlanner<'a> {
+    pub fn new(ctx: &'a SessionContext) -> Self {
+        Self { ctx }
+    }
+}
+
+#[async_trait]
+impl<'a> PhysicalPlanner for CustomPhysicalPlanner<'a> {
```

Comment:
Should be able to implement `ExtensionPlanner` to avoid needing to implement all the nodes: https://github.com/apache/arrow-datafusion/blob/e253b8e505d22dc4b427109917470db39aa3815b/datafusion/core/src/physical_planner.rs#L395-L416

Reply:
I was thinking about that. I think we should try to start with the `ExtensionPlanner`. We'll likely end up having to do a completely custom physical planner at some point. But if we can get away with the ExtensionPlanner for now, that'd be great!